relay-server/src/actors/store.rs

```diff
@@ -199,46 +199,102 @@ impl StoreService {
-                attachments,
-            });
+        if event_item.is_none() && attachments.is_empty() {
+            // No event-related content. All done.
+            return Ok(());
+        }
 
-            self.produce(topic, scoping.organization_id, event_message)?;
-            metric!(
-                counter(RelayCounters::ProcessingMessageProduced) += 1,
-                event_type = &event_item.ty().to_string()
-            );
-        } else if !attachments.is_empty() {
-            relay_log::trace!("Sending individual attachments of envelope to kafka");
-            for attachment in attachments {
-                let attachment_message = KafkaMessage::Attachment(AttachmentKafkaMessage {
-                    event_id: event_id.ok_or(StoreError::NoEventId)?,
-                    project_id: scoping.project_id,
-                    attachment,
-                });
+        let remote_addr = envelope.meta().client_addr().map(|addr| addr.to_string());
+
+        let kafka_messages = Self::extract_kafka_messages_for_event(
+            event_item,
+            event_id.ok_or(StoreError::NoEventId)?,
+            scoping,
+            start_time,
+            remote_addr,
+            attachments,
+        );
+
+        for message in kafka_messages {
+            // This line is prior to self.produce as it takes `message` by value.
+            let is_attachment = matches!(&message, KafkaMessage::Attachment(_));
 
-                self.produce(topic, scoping.organization_id, attachment_message)?;
+            self.produce(topic, scoping.organization_id, message)?;
+
+            // If message was succesfully sent, make the right log message.
+            if is_attachment {
                 metric!(
                     counter(RelayCounters::ProcessingMessageProduced) += 1,
                     event_type = "attachment"
                 );
+            } else if let Some(event_item) = event_item {
+                metric!(
+                    counter(RelayCounters::ProcessingMessageProduced) += 1,
+                    event_type = &event_item.ty().to_string()
+                );
             }
         }
 
         Ok(())
     }
 
+    fn extract_kafka_messages_for_event(
+        event_item: Option<&Item>,
+        event_id: EventId,
+        scoping: Scoping,
+        start_time: Instant,
+        remote_addr: Option<String>,
+        attachments: Vec<ChunkedAttachment>,
+    ) -> impl Iterator<Item = KafkaMessage> {
+        // There might be a better way to do this:
+        let (individual_attachments, inline_attachments) = match event_item {
+            Some(event_item) => {
+                if matches!(event_item.ty(), ItemType::Transaction) {
+                    // Sentry discards inline attachments for transactions, so send them as individual ones instead.
+                    relay_log::trace!("Sending transaction event item of envelope to kafka");
```

Comment:
Thoughts on putting log messages here?

Reply:
Would we wanna log a message for every single attachment sent then? previously we just logged that we were about to send attachments as a single log message